BUG: test_mypy_example[mongodb] fails

### What pyzmq version?

25.0.2

### What libzmq version?

4.3.4

### Python version (and how it was installed)

3.9.16; pkg install runtime/python

### OS

OpenIndiana Hipster

### What happened?

I'm packaging `pyzmq` for OpenIndiana and during testing I noticed one test fails:
```
__________________________ test_mypy_example[mongodb] __________________________

example = 'mongodb'

    @pytest.mark.skipif(
        not os.path.exists(examples_dir), reason="only test from examples directory"
    )
    @pytest.mark.parametrize("example", examples)
    def test_mypy_example(example):
        example_dir = os.path.join(examples_dir, example)
>       run_mypy("--disallow-untyped-calls", example_dir)

zmq/tests/test_mypy.py:64:  
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

mypy_args = ('--disallow-untyped-calls', 'examples/mongodb')
p = <subprocess.Popen object at 0x7fffa3579d50>
o = b'examples/mongodb/controller.py:35: error: Module has no attribute "Connection"  [attr-defined]\nFound 1 error in 1 file (checked 2 source files)\n'
_ = None

    def run_mypy(*mypy_args):
        """Run mypy for a path
    
        Captures output and reports it on errors
        """
        p = Popen(
            [sys.executable, "-m", "mypy"] + list(mypy_args), stdout=PIPE, stderr=STDOUT
        )
        o, _ = p.communicate()
        out = o.decode("utf8", "replace")
        print(out)
>       assert p.returncode == 0, out
E       AssertionError: examples/mongodb/controller.py:35: error: Module has no attribute "Connection"  [attr-defined]
E         Found 1 error in 1 file (checked 2 source files)
E
E       assert 1 == 0
E        +  where 1 = <subprocess.Popen object at 0x7fffa3579d50>.returncode

zmq/tests/test_mypy.py:47: AssertionError
----------------------------- Captured stdout call -----------------------------
examples/mongodb/controller.py:35: error: Module has no attribute "Connection"  [attr-defined]
Found 1 error in 1 file (checked 2 source files)
```

### Code to reproduce bug

```python
Run pytest.
```


### Traceback, if applicable

_No response_

### More info

_No response_

--- a/examples/mongodb/controller.py
+++ b/examples/mongodb/controller.py
@@ -9,7 +9,7 @@ import json
 from typing import Any, Dict, Optional, Union
 
 import pymongo
-import pymongo.json_util
+from bson import json_util
 
 import zmq
 
@@ -32,12 +32,12 @@ class MongoZMQ:
         self._bind_addr = bind_addr
         self._db_name = db_name
         self._table_name = table_name
-        self._conn = pymongo.Connection()
+        self._conn: pymongo.MongoClient = pymongo.MongoClient()
         self._db = self._conn[self._db_name]
         self._table = self._db[self._table_name]
 
     def _doc_to_json(self, doc: Any) -> str:
-        return json.dumps(doc, default=pymongo.json_util.default)
+        return json.dumps(doc, default=json_util.default)
 
     def add_document(self, doc: Dict) -> Optional[str]:
         """
@@ -50,7 +50,7 @@ class MongoZMQ:
             return 'Error: %s' % e
         return None
 
-    def get_document_by_keys(self, keys: Dict[str, Any]) -> Union[Dict, str]:
+    def get_document_by_keys(self, keys: Dict[str, Any]) -> Union[Dict, str, None]:
         """
         Attempts to return a single document from database table that matches
         each key/value in keys dictionary.

--- a/test-requirements.txt
+++ b/test-requirements.txt
@@ -6,6 +6,7 @@ coverage<5
 flake8
 gevent; platform_python_implementation != "PyPy" and sys_platform != "win32" and sys_platform != "darwin" and python_version < "3.11"
 mypy; platform_python_implementation != "PyPy"
+pymongo
 pytest
 pytest-asyncio>=0.16; python_version < "3.7"
 pytest-asyncio>=0.17; python_version >= "3.7"

